.github/workflows/go.yml

```diff
@@ -3,8 +3,11 @@ name: Go
 on:
   push:
     branches: [main]
+    paths-ignore:
+      - '**.md'
   pull_request:
-    branches: [main]
```

Comment:
I don't see a reason to limit the tests to only run when a PR is opened against the main branch. Removing this restriction will enable the development of feature branches and opening PRs against them.